pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/AuthenticationTest.java

```diff
@@ -1,110 +1,129 @@
 package org.broadinstitute.dsm;
 
-import static org.broadinstitute.dsm.TestHelper.cfg;
-import static org.broadinstitute.dsm.TestHelper.setupDB;
-
-import org.broadinstitute.dsm.db.dao.user.UserDao;
-import org.broadinstitute.dsm.db.dto.user.UserDto;
 import org.broadinstitute.dsm.model.NameValue;
 import org.broadinstitute.dsm.model.patch.Patch;
+import org.broadinstitute.dsm.service.admin.UserAdminTestUtil;
 import org.broadinstitute.dsm.statics.DBConstants;
 import org.broadinstitute.dsm.util.UserUtil;
+import org.junit.AfterClass;
 import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Ignore;
+import org.junit.BeforeClass;
 import org.junit.Test;
 
-public class AuthenticationTest {
-    public static final String UNIT_TESTER_EMAIL = "testUsersEmails.unitTesterEmail";
-    public static final String GP_UNIT_TESTER_EMAIL = "testUsersEmails.gpUnitTesterEmail";
+import java.util.Arrays;
+import java.util.Collections;
 
+import static org.broadinstitute.dsm.service.admin.UserAdminService.USER_ADMIN_ROLE;
+import static org.broadinstitute.dsm.statics.DBConstants.KIT_SHIPPING;
+import static org.broadinstitute.dsm.statics.DBConstants.PT_LIST_VIEW;
 
-    @Before
-    public void first() {
-        setupDB();
+public class AuthenticationTest extends DbTxnBaseTest {
 
-    }
-    @Test
-    @Ignore
-    public void GPPatchKitAccessTest(){
-        String realm = "osteo2";
+    private static String dsmAdminUserId;
 
-        String gpUserId = Integer.toString(new UserDao().getUserByEmail(cfg.getString(GP_UNIT_TESTER_EMAIL)).orElse(new UserDto()).getId());
+    private static String cmiKitShippingOnlyUserId;
 
-        Patch patch = new Patch("0",  "dsmKitRequestId", "0", gpUserId , new NameValue("kit.collectionDate",  "2023-04-24"), null, "XSZSRS1MS3D4OAEK2DPM") ;
-        patch.setTableAlias("kit");
+    private static final UserAdminTestUtil cmiAdminUtil = new UserAdminTestUtil();
+
+    private static String studyInstanceName;
 
-        Assert.assertTrue(UserUtil.checkKitShippingAccessForPatch(realm, gpUserId, null, patch));
+    private static String generateUserEmail() {
+        return "AuthTest-" + System.currentTimeMillis() + "@broad.dev";
+    }
+
+    /**
+     * Create new groups, a new ddp_instance, associated roles,
+     * a few new users and set their permissions.
+     */
+    @BeforeClass
+    public static void setup() {
+        String nameAppend = "." + System.currentTimeMillis();
+        String studyInstanceName = "instance." + nameAppend;
+        String cmiStudyGroup = "cmi" + nameAppend;
+
+        cmiAdminUtil.createRealmAndStudyGroup(studyInstanceName, cmiStudyGroup);
+        cmiAdminUtil.setStudyAdminAndRoles(generateUserEmail(), USER_ADMIN_ROLE, Arrays.asList(KIT_SHIPPING, PT_LIST_VIEW));
```

Comment:
This returns the admin user ID. The line below is unnecessary (and since it creates a different email address there might be unexpected results if the test is expanded).

Reply:
If I set `dsmAdminUserId` to the return from L45, one of the tests fails.  So I think we do need L47.

Reply:
Ah, I see the issue. setStudyAdminAndRoles set the study roles but does not assign them to the admin. I could probably fix that (a bit of an assumption), but perhaps I was mostly confused by the terminology: dsmAdminUserId in this case is not a user admin, but a regular user.

Reply:
Will change the name of var to be more clear.